easybuild/easyblocks/f/ferret.py

```diff
@@ -48,80 +53,156 @@ def configure_step(self):
         """Configure Ferret build."""
 
         buildtype = "x86_64-linux"
-
-        try:
-            os.chdir('FERRET')
-        except OSError, err:
-            raise EasyBuildError("Failed to change to FERRET dir: %s", err)
+        if LooseVersion(self.version) < LooseVersion("7.3"):
+            try:
+                os.chdir('FERRET')
+            except OSError, err:
+                raise EasyBuildError("Failed to change to FERRET dir: %s", err)
 
         deps = ['HDF5', 'netCDF', 'Java']
 
         for name in deps:
             if not get_software_root(name):
                 raise EasyBuildError("%s module not loaded?", name)
 
-        fn = "site_specific.mk"
+        if LooseVersion(self.version) >= LooseVersion("7.3"):
+            try:
+                shutil.copy2('external_functions/ef_utility/site_specific.mk.in',
+                             'external_functions/ef_utility/site_specific.mk')
+                shutil.copy2('site_specific.mk.in', 'site_specific.mk')
+            except OSError, err:
+                raise EasyBuildError("Failed to copy external_functions/ef_utility/site_specific.mk.in " +
+                                     "to external_functions/ef_utility/site_specific.mk or site_specific.mk.in " +
+                                     "to site_specific.mk: %s", err)
+            fns = [
+                "site_specific.mk",
+                "external_functions/ef_utility/site_specific.mk",
+            ]
+        else:
+            fns = "site_specific.mk"
 
-        for line in fileinput.input(fn, inplace=1, backup='.orig'):
+        for fn in fns:
+            for line in fileinput.input(fn, inplace=1, backup='.orig'):
```

Comment:
@hajgato Would you mind updating this to use the `apply_regex_substitutions` function from `filetools`?